### PR TITLE
Keyboard controller

### PR DESCRIPTION
**Pull request by saishravanm**
Completed rudimentary Keyboard Controller Node. Located in /src/tools/keyboard_controller. The code is simple enough that it can be tweaked easily, but in the future, hoping to use ROS parameters to set custom parameters for speed, throttle, etc. 

Controls: 
W - Throttle up 
S - Throttle down 
A - Go left 
D - Go right 
Hold X - Reverse
Hold X + W - Go backward

### CODE REVIEW

**Comment by jruths** [comment]
Ah!  This needs to be a pull request into dev, not main!

**Comment by saishravanm** [comment]
@jruths Fixed!

File: src/tools/keyboard_control/test/test_copyright.py
Comment (akevinge) [inline src/tools/keyboard_control/test/test_copyright.py:1]: Let's delete this test if it's unused.

File: src/tools/keyboard_control/test/test_flake8.py
Comment (akevinge) [inline src/tools/keyboard_control/test/test_flake8.py:1]: As previously commented.

File: src/tools/keyboard_control/test/test_pep257.py
Comment (akevinge) [inline src/tools/keyboard_control/test/test_pep257.py:1]: Same for this.

File: src/tools/keyboard_control/keyboard_control/keyboard_control.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        elif pygame.K_s in self.pressed_keys:
+            self.throttle -= 0.05
+            self.brake += 0.05
+            print("Braking! \n")
+        elif pygame.K_a in self.pressed_keys:
+            self.steer -= 0.03
+            print("Going left! \n")
+        elif pygame.K_d in self.pressed_keys:
+            self.steer += 0.03
+            print("Going right!\n")
```
Comment (akevinge) [inline src/tools/keyboard_control/keyboard_control/keyboard_control.py:53]: Do we need these print statements or are they just for debugging. If they're just for debugging, we should probably remove them.

File: src/tools/keyboard_control/keyboard_control/keyboard_control.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                self.pressed_keys.add(event.key)
+            if event.type == pygame.KEYUP:
+                self.pressed_keys.discard(event.key)
+                
+        self.handle_controls()
+    
+
+def main(args=None):
+    rclpy.init(args = args)
+    minimal_publisher = KeyboardController()
```
Comment (akevinge) [inline src/tools/keyboard_control/keyboard_control/keyboard_control.py:80]: Nit: rename this to something that makes more sense.

File: src/tools/keyboard_control/keyboard_control/keyboard_control.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        
+    #publishes vehicle information to the vehicle control topic
+    def control_callback(self):
+        control_msg = VehicleControl()
+        control_msg.throttle = self.throttle
+        control_msg.brake = self.brake 
+        control_msg.steer = self.steer
+        control_msg.reverse = self.reverse
+        self.command_publisher.publish(control_msg)
+        #print("Published Message!\n")
```
Comment (akevinge) [inline src/tools/keyboard_control/keyboard_control/keyboard_control.py:37]: Debug statement left in here.

**Review by akevinge — CHANGES_REQUESTED** [review]

**Review by akevinge — APPROVED** [review]

**Comment by akevinge** [comment]
Looks like the pre-compiled download URL for lemon is down. Feel free to merge despite the failing checks :+1:

**Pull request merged**